tile-generation/src/main/scala/com/oculusinfo/tilegen/tiling/TileOperations.scala

```diff
@@ -0,0 +1,353 @@
+    val colExprs = colSpecs.map(colSpecToExpression(input.sqlContext, _))
+    val result = input.srdd.select(colExprs:_*)
+    PipelineData(input.sqlContext, result)
+  }
+
+  /**
+   * Parse args for basic heatmap tile generator.
+   */
+  def parseHbaseHeatmapOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val zookeeperQuorum = argParser.getString("hbase.zookeeper.quorum", "Zookeeper quorum addresses")
+    val zookeeperPort = argParser.getString("hbase.zookeeper.port", "Zookeeper quorum addresses")
+    val hbaseMaster = argParser.getString("hbase.master", "Zookeeper quorum addresses")
+    val heatmapParams = parseHeatMapOpImpl(args, argParser)
+
+    hbaseHeatMapOp(heatmapParams._1, heatmapParams._2, heatmapParams._3, zookeeperQuorum, zookeeperPort, hbaseMaster)_
+  }
+
+  private def parseHeatMapOpImpl(args: Map[String, String], argParser: KeyValueArgumentSource) = {
+    val xColSpec = argParser.getString("ops.xColumn", "Colspec denoting data x column")
+    val yColSpec = argParser.getString("ops.yColumn", "Colspec denoting data y column")
+
+    val taskParametersFactory = new TilingTaskParametersFactory(null, List("ops"))
+    taskParametersFactory.readConfiguration(JsonUtilities.mapToJSON(args))
+    val taskParameters = taskParametersFactory.produce(classOf[TilingTaskParameters])
+
+    (xColSpec, yColSpec, taskParameters)
+  }
+
+  /**
+   * A basic heatmap tile generator that writes output to HDFS.
+   */
+  def hbaseHeatMapOp(xColSpec: String,
+                yColSpec: String,
+                tilingParms: TilingTaskParameters,
+                zookeeperQuorum: String,
+                zookeeperPort: String,
+                hbaseMaster: String)
+               (input: PipelineData) = {
+
+    val tileIO = new HBaseTileIO(zookeeperQuorum, zookeeperPort, hbaseMaster)
+    heatMapOpImpl(xColSpec, yColSpec, tilingParms, tileIO)(input)
+  }
+
+  /**
+   * Parse args for basic heatmap tile generator.
+   */
+  def parseFileHeatmapOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val heatmapParams = parseHeatMapOpImpl(args, argParser)
+    fileHeatMapOp(heatmapParams._1, heatmapParams._2, heatmapParams._3)_
+  }
+
+  /**
+   * A basic heatmap tile generator that writes output to the local file system.
+   */
+  def fileHeatMapOp(xColSpec: String,
+                yColSpec: String,
+                tilingParms: TilingTaskParameters)
+               (input: PipelineData) = {
+    val tileIO = new LocalTileIO(".avro")
+    heatMapOpImpl(xColSpec, yColSpec, tilingParms, tileIO)(input)
+  }
+
+  private def heatMapOpImpl(xColSpec: String,
+                    yColSpec: String,
+                    tilingParms: TilingTaskParameters,
+                    tileIO: TileIO)
+                   (input: PipelineData) = {
+    // TODO: Switch to the factory based invocation
+    val valuer = new CountValueExtractor[Double, java.lang.Double]()
```

Comment:
I'm not sure if it's a useful thing or not - do we want to switch this to an Int or Long instead of Double, if we are mandating a count?
Pro: Int arithmetic should be quicker
Con: I think it actually is slower these days, weirdly enough